clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/cache/client/IamRoleCacheClient.java

```diff
@@ -23,33 +23,38 @@
 import org.springframework.beans.factory.annotation.Autowired;
 
 import java.util.Collection;
+import java.util.Collections;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.stream.Collectors;
 
 import static com.netflix.spinnaker.clouddriver.ecs.cache.Keys.Namespace.IAM_ROLE;
 
-public class IamRoleCacheClient {
-
-  private final Cache cache;
+public class IamRoleCacheClient extends AbstractCacheClient<IamRole>{
 
   @Autowired
-  public IamRoleCacheClient(Cache cache) {
-    this.cache = cache;
+  public IamRoleCacheClient(Cache cacheView) {
+    super(cacheView, IAM_ROLE.toString());
+  }
+
+  @Override
+  public IamRole get(String key) {
```

Comment:
Is this method even used?  Do we ever get a single role?

Also, I don't like the `iterator().next()` part.  I don't have a suggestion there though.  That being said, I am not even sure this method is necessary for IAM roles

Reply:
It's not ever used, but you get it part of the `AbstractCacheClient` as every other subclass.

I will actually remove that method as it doesn't make sense to override it - a specific key should return only the cache data for that key, there cannot be multiple cache data for 1 key.